src/common/dav/dav.h

```diff
@@ -122,8 +117,16 @@ typedef int (*dav_constr)(dav_obj_t *pop, void *ptr, void *arg);
  * initialized, or if it's interrupted before the constructor completes, the
  * memory reserved for the object is automatically reclaimed.
  */
-int dav_alloc(dav_obj_t *pop, uint64_t *offp, size_t size,
-	      uint64_t type_num, dav_constr constructor, void *arg);
+int
+dav_alloc(dav_obj_t *pop, uint64_t *offp, size_t size, uint64_t type_num, dav_constr constructor,
+	  void *arg);
+
+/*
+ * Allocates with flags a new object from the pool.
```

Comment:
As far as I can see, `dav_xalloc()` does exactly the same as `dav_alloc()` does except for accepting additional flags. I hate duplication so, maybe something like this will suffice:

```
Does what dav_alloc() does but accepts additional flags.
```

At some point, documenting the meaning of these flags should enrich this documentation.

Reply:
x stands for extended alloc. Here we pass a 64 bit flag as additional variable. The flag embeds info on the allocation class to be used, bucket/zone in which the allocation should happen and additional bits like zero the allocated memory etc. This is a convention taken from pmdk.

Reply:
I did some cleanup. I am keeping only the extended version of the api and renamed without x in it.